src/fifo/FifoBuffer.cpp

```diff
@@ -101,14 +103,22 @@ int32_t FifoBuffer::read(void *buffer, int32_t numFrames) {
     if ((readIndex + framesToRead) > mFrameCapacity) {
         // read in two parts, first part here is at the end of the mStorage buffer
         uint32_t frames1 = mFrameCapacity - readIndex;
-        int32_t numBytes = convertFramesToBytes(frames1);
+        int32_t numBytes = convertFramesToBytes(frames1); // Implicitly casting
+        // `uint32_t` (frames1) to `int32_t`? Same on line 102.
         if (numBytes < 0) {
             return static_cast<int32_t>(Result::ErrorOutOfRange);
         }
         memcpy(destination, source, static_cast<size_t>(numBytes));
         destination += numBytes;
         // read second part, which is at the beginning of mStorage
-        source = &mStorage[0];
+        source = &mStorage[0];  // I'm not sure I udnerstand this. Let's say,
+        // `readIndex` is 0, but `framesToRead > mFrameCapacity`. In this case,
```

Comment:
We should clip framesToRead to mFramesCapacity.

Reply:
We clip framesToRead to framesAvailable, which should be enough. This is assuming that mFifo->getFullFramesAvailable() is never bigger than mFrameCapacity, which would make sense.

Reply:
Fixed in #603